nipype/interfaces/base/specs.py

```diff
@@ -309,6 +311,11 @@ def _get_sorteddict(self,
                     out = objekt
         return out
 
+    @property
+    def __all__(self):
+        return [k for k,v in self.items() if not k == "__all__"]
```

Comment:
Given that `items()` provides a dict keyed by `self.copyable_trait_names()`, and you only need keys, I think you can just make this:

```Python
@property
def __all__(self):
    return self.copyable_trait_names()
```

You'll need to change the BET tests to either sort or check set equality, but this should work.